src/main/java/com/softserve/teamproject/config/WebSecurityConfig.java

```diff
@@ -9,11 +16,50 @@
 
 @Configuration
 @EnableWebSecurity
-public class WebSecurityConfig  extends WebSecurityConfigurerAdapter {
+public class WebSecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private UserDetailsServiceImpl userDetailsService;
+
     @Override
     protected void configure(HttpSecurity http) throws Exception {
         http
                 .authorizeRequests()
-                .antMatchers("/", "/index").permitAll();
+                .antMatchers("/welcome").authenticated()
+                .and()
+                .csrf().disable()
+                .authorizeRequests()
+                .and()
+                .formLogin()//.loginPage("/signin").permitAll()
+                .defaultSuccessUrl("/welcome")
+                .usernameParameter("username")
+                .passwordParameter("password")
+                .failureUrl("/signin-error")
+                .and()
+                .logout().logoutUrl("/logout").logoutSuccessUrl("/")
+                .and()
+                .rememberMe().key("token").tokenValiditySeconds(3600);
+    }
+
+////We don't need a password encoder while we don't have registration
+//    @Bean
+//    public PasswordEncoder passwordEncoder() {
+//        return new BCryptPasswordEncoder();
+//    }
+
+    @Override
+    protected void configure(AuthenticationManagerBuilder auth)
+            throws Exception {
```

Comment:
Ooops, sorry)) I thought you were speaking about all the configure() methods (they are overloaded in that class). And in one of them it is possible to use more specific exception type. Thank you)